### PR TITLE
[IndexFilters] consolidate se23 styles and logic 

### LINKED ISSUE #9938: [IndexFilters] Consolidate se23 logic and styles

### PR DESCRIPTION
**Pull request by gwyneplaine**
### WHY are these changes introduced?

Fixes #9938 

### WHAT is this pull request doing?

* Consolidate se23 styles 
* Consolidate se23 logic
* Remove custom `FilterButton` and replace it with direct invocations of the Polaris Button instead. This seems safe to do because:
  *  even though we export `FilterButton` from the root of `IndexFilters`, we do not actually expose it directly from the `polaris-react` [entrypoint](https://github.com/Shopify/polaris/blob/2c718dc3ecf9393119cd1e08c387b648db78449a/polaris-react/src/index.ts#L190).
  * AFAIK, noone is reaching into our cjs/esm builds to grab`FilterButton` (see [grokt](https://grokt.shopify.io/results?q=%40shopify%2Fpolaris%2F.*%2Fcomponents%2FIndexFilters))
  * Just to be sure, also doesn't look like anyone is importing `FilterButton` from @shopify/polaris either.  (see [grokt](https://grokt.shopify.io/results?q=FilterButton)) 
  * That said I can see an argument being made for providing convenience wrappers to alleviate repetitive config, so deferring to the team. :+1:  Happy to revert if necessary 👍 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Review by laurkim — APPROVED** [review]
Nice 🚀 

Small issue with spacing of filter and sort buttons on xs and sm breakpoints but other than that, looks great 🙌 
abc25-14-nhd1z-kug3z

**Review by sophschneider — APPROVED** [review]
Nice! removing the custom button is good w me :) Can you add a changeset for it?

**Review by sam-b-rose — APPROVED** [review]

**Comment by gwyneplaine** [comment]
> Nice 🚀
> 
> Small issue with spacing of filter and sort buttons on xs and sm breakpoints but other than that, looks great 🙌 abc25-14-nhd1z-kug3z

Looks to be an inadvertent bugfix. New spacing is true to [Figma](https://www.figma.com/file/jLLkmo9r28hiqPvf4s90E4/Polaris-Gen-3-Components-%5Balpha%5D?type=design&node-id=67128-21676&mode=dev) 
Screenshot 2023-08-28 at 2 12 26 pm

Production: 
Screenshot 2023-08-28 at 2 14 16 pm

File: polaris-react/src/components/IndexFilters/IndexFilters.scss [hunk 1]
```diff
@@ -30,22 +30,17 @@ $spinner-size: 20px;
...
+  @media #{$p-breakpoints-md-down} {
+    height: var(--p-space-12);
   }
 }
 
 .SmallScreenTabsWrapper {
   overflow: hidden;
   padding: var(--p-space-1) var(--p-space-0) var(--p-space-2) var(--p-space-3);
   padding: 0;
-  height: auto;
```
Comment (gwyneplaine) [inline polaris-react/src/components/IndexFilters/IndexFilters.scss:48]: This was overriding the `.TabsWrapper` height invocation on mdDown above (`line 36`). 
As this class is only applied at `mdDown`, safe for us to remove this duplicative height rule.

**Pull request merged**